### PR TITLE
Exclude the ``_ignore_`` attribute from the ``__members__`` container.

### PR DESCRIPTION
**Pull request by mbyrnepr2**
Closes pylint-dev/pylint#9015

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
|    | :scroll: Docs          |

## Description
Exclude the ``_ignore_`` attribute from the ``__members__`` container.

Closes [#9015](https://github.com/pylint-dev/pylint/issues/9015)

### CODE REVIEW

**Review by Pierre-Sassoulas — COMMENTED** [review]
Look like we might need to also exclude ``_missing_``?  https://docs.python.org/3/library/enum.html

**Comment by mbyrnepr2** [comment]
`_missing_` is a class method so Pylint is already happy with that naming. What do you think @Pierre-Sassoulas ?

**Review by Pierre-Sassoulas — COMMENTED** [review]
😄 Hmm, and what about ``_name_`` for example ? Just stumbled upon https://docs.python.org/3/library/enum.html#supported-sunder-names (and learnt about sunder vs dunder, nice) it doesn't seem to be all functions in this case.

**Comment by Pierre-Sassoulas** [comment]
Thank you for investigating ! If I understand correctly ``_value_`` is somewhat similar and ``_names_`` is not ? (It seems ``ValueError: _names_ are reserved for future Enum use`` should raise an error message in pylint, but it's very specific to enums, I don't know if we should do it and what message to raise.) If that's the case we could do ``_ignore_`` and ``_value_`` together and ``_names_`` later ? If not let's just do ``_ignore_`` to not overcomplicate things.

**Comment by mbyrnepr2** [comment]
Ah my example made things confusing I think :)
The Python output `_names_ are reserved for future Enum use` is referring to sunder names in general (`_names_`).
Also `_value_` and `_name_` do produce the `invalid-name`  so I'll see about handling that here.


`ValueError: _names_ are reserved for future Enum use` could be a separate issue I think.

File: ChangeLog [hunk 1]
```diff
@@ -221,6 +223,11 @@ Release date: TBA
 
   Closes pylint-dev/pylint#8802
 
+
+* Exclude the ``_ignore_`` attribute from the ``__members__`` container.
+
+  Closes pylint-dev/pylint#9015
```
Comment (mbyrnepr2) [inline ChangeLog:229]:
```suggestion
* Fix false positives for ``no-member`` and ``invalid-name`` when using the ``_name_``, ``_value_`` and ``_ignore_`` sunders in Enums.

  Closes pylint-dev/pylint#9015
```

File: ChangeLog [hunk 2]
```diff
@@ -22,6 +22,8 @@ Release date: TBA
 * Exclude class attributes from the ``__members__`` container of an ``Enum`` class when they are
   ``nodes.AnnAssign`` nodes with no assigned value.
 
+* Fix a false positive for ``no-member`` when accessing the ``_name_`` and ``_value_`` sunders of an Enum member value.
+
```
Comment (mbyrnepr2) [inline ChangeLog:26]:
```suggestion
```

**Review by Pierre-Sassoulas — APPROVED** [review]
Great !

**Comment by Pierre-Sassoulas** [comment]
Maybe we shouldn't close the pylint issue until astroid is released upgraded and we could clean-up the history and use two commits instead of a squash ? Or we can squash, I'll let you decide :)

**Comment by Pierre-Sassoulas** [comment]
We actually need to squash for the backport job to work automatically

**Comment by jacobtylerwalls** [comment]
Looks like this wasn't backported, so I'll move all the labels and such here and on the pylint issue. Life goes on. :D

**Pull request merged**